src/main/java/nl/andrewl/emaildatasetbrowser/view/search/export/QueryExportParameterPanel.java

```diff
@@ -0,0 +1,117 @@
+
+import nl.andrewl.email_indexer.data.export.query.QueryExportParams;
+import nl.andrewl.email_indexer.data.export.query.QueryExporter;
+import nl.andrewl.emaildatasetbrowser.view.search.LuceneSearchPanel;
+
+/**
+ * Simple JPanel base class for Exporting queries.
+ */
+public abstract class QueryExportParameterPanel extends JPanel {
+    private LuceneSearchPanel searchPanel;
+
+    private final JSpinner maxResultsSpinner = new JSpinner(
+            new SpinnerNumberModel(100, 1, 10000, 1));
+    private final JCheckBox separateThreadsToggle = new JCheckBox("Separate mailing threads");
+    private final JTextField outputPathField = new JTextField();
+
+    private Path outputPath = null;
+
+    public QueryExportParameterPanel(LuceneSearchPanel searchPanel) {
+        super(new FlowLayout(FlowLayout.LEFT));
+        this.searchPanel = searchPanel;
+        setLayout(new BoxLayout(this, BoxLayout.PAGE_AXIS));
+        setVisible(false);
+
+        add(new JLabel("Max. result count:"));
+        add(maxResultsSpinner);
+
+        JButton searchDirectoryButton = new JButton("Select Directory");
+        JButton searchFileButton = new JButton("Select File");
+        separateThreadsToggle.addActionListener(e -> {
+            searchDirectoryButton.setVisible(separateThreadsToggle.isSelected());
+            searchFileButton.setVisible(!separateThreadsToggle.isSelected());
+            outputPath = null;
+            updatePathField();
+        });
+        add(separateThreadsToggle);
+
+        searchDirectoryButton.addActionListener(e -> {
+            JFileChooser fc = new JFileChooser();
+            fc.setFileSelectionMode(JFileChooser.DIRECTORIES_ONLY);
+            if (fc.showSaveDialog(this) != JFileChooser.APPROVE_OPTION) {
+                return;
+            }
+            this.outputPath = fc.getSelectedFile().toPath();
+            updatePathField();
+        });
+        searchDirectoryButton.setVisible(false);
+        add(searchDirectoryButton);
+
+        searchFileButton.addActionListener(e -> {
+            JFileChooser fc = new JFileChooser();
+            fc.setFileSelectionMode(JFileChooser.FILES_ONLY);
+            fc.setFileFilter(buildFileNameFilter());
+            fc.setAcceptAllFileFilterUsed(true);
+            if (fc.showSaveDialog(this) != JFileChooser.APPROVE_OPTION) {
+                return;
+            }
+            this.outputPath = fc.getSelectedFile().toPath();
+            updatePathField();
+        });
+        add(searchFileButton);
+
+        outputPathField.setEditable(false);
+        add(outputPathField);
+    }
+
+    /**
+     * Updates the displayed path.
+     */
+    private void updatePathField() {
+        if (this.outputPath != null) {
+            this.outputPathField.setText(this.outputPath.toString());
+        } else {
+            this.outputPathField.setText("");
+        }
+    }
+
+    /**
+     * Exports the query results.
+     */
+    public CompletableFuture<Void> export() {
```

Comment:
I would suggest to maybe add a validation step before calling export, so that you don't have to just fall back to handling an exception if the query is empty or the output path isn't set or something like that.